tree/ntuple/v7/doc/specifications.md

```diff
@@ -501,6 +554,43 @@ The frame hierarchy is as follows
     |---- Column group 2 record frame
     | ...
 
+
+#### Cluster Group Record Frame
+
+The cluster group record frame references the page list envelopes for groups of clusters.
+A cluster group record frame starts with
+
+```
+ 0                   1                   2                   3
+ 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|                                                               |
++                      Minimum Entry Number                     +
+|                                                               |
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|                                                               |
++                          Entry Span                           +
+|                                                               |
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|                       Number of clusters                      |
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+```
+Followed by the page list envelope link.
+
+The minimum entry number is the minimum first entry of the clusters in the cluster group.
+The entry span is the number of entries that are (partially for sharded clusters) covered by this cluster group.
+The entry range allows for finding the right page list for random access requests to entries.
```

Comment:
>random access

at some point Enrico (?) told me in order to not read all `page`s within a `cluster`, RNTuple iterator in ROOT basically only support iterate and read new `page` on demand.

My feeling is that hasn't changed yet, so does "random access" here referring to you can jump to the "right page list" but you still have to do sequential search through pages within that list in order to find event?